src/screens/AddEditObservation/ObservationBase.js

```diff
@@ -107,6 +107,31 @@ const ObservationBase = props => {
     navigation.setParams({ isDeclineChangesPopupOpened: false });
     navigation.goBack();
   };
+
+  useEffect(() => {
+    const isDeclineChangesSet = () =>
+      navigation.state.params &&
+      navigation.state.params.isDeclineChangesPopupOpened;
+
+    const onBackButtonPressAndroid = () => {
+      if (isDeclineChangesSet()) {
+        // returning false to prevent android back button event
+        return false;
+      }
+      navigation.setParams({ isDeclineChangesPopupOpened: true });
+      return true;
+    };
+
+    BackHandler.addEventListener("hardwareBackPress", onBackButtonPressAndroid);
+
+    return () => {
+      BackHandler.removeEventListener(
+        "hardwareBackPress",
+        onBackButtonPressAndroid
+      );
+    };
+  });
```

Comment:
does it makes sense to define dependencies as a second parameter, so we won't trigger this hook on every rerender?